packages/compile-common/src/compilerSupplier/strategy.ts

```diff
@@ -0,0 +1,106 @@
+import type { Constructor as FreeConstructor } from "./constructor";
+import type { Results as FreeResults } from "./results";
+
+export namespace Strategy {
+  /**
+   * Type-level description of a compiler supplier strategy
+   */
+  export type Specification = {
+    constructor: Omit<FreeConstructor.Specification, "result">;
+    results: FreeResults.Specification;
+    allowsLoadingSpecificVersion: boolean;
+    allowsListingVersions: boolean;
+  };
+
+  /**
+   * Scoped re-exports of Constructor, using Strategy.Specification as
+   * generic param
+   */
+  export namespace Constructor {
+    /**
+     * Getter for the specified constructor, intersected with fixed result
+     * type of Strategy<S>
+     */
+    export type Specification<S extends Strategy.Specification> =
+      S["constructor"] & { result: Strategy<S>; };
+
+    /**
+     * Constructor options argument for a specified strategy
```

Comment:
Probably worth explicitly noting that this is just a re-export (so people don't go wasting time trying to figure out what it's doing).

Reply:
explicitly noted.